### ISSUE TITLE
Mistake in type hinting Exponential Smoothing model - seasonality parameter 

### ISSUE THREAD

**Issue by rijkvandermeulen** [unit8co/darts#1185]
I believe the type hinting for the seasonal parameter in the Exponential Smoothing model is incorrect (ModelMode iso SeasonalityMode): 

```
class ExponentialSmoothing(ForecastingModel):
    def __init__(
        self,
        trend: Optional[ModelMode] = ModelMode.ADDITIVE,
        damped: Optional[bool] = False,
        seasonal: Optional[ModelMode] = SeasonalityMode.ADDITIVE,
        seasonal_periods: Optional[int] = None,
        random_state: int = 0,
        **fit_kwargs,
    ):
```
I'll open a PR to make this quick fix. :)

BTW: have you ever considered adding mypy to your pipeline  and pre-commit hooks to spot issues like this?

**Comment by gdevos010** [comment]
I can't speak for @hrzn, but I personally would love to see mypy added to the pre-commit hooks and pipeline. Mypy currently shows 119 errors which will need to be resolved first. Would you be able to help fix those?

**Comment by hrzn** [comment]
> I can't speak for @hrzn, but I personally would love to see mypy added to the pre-commit hooks and pipeline. Mypy currently shows 119 errors which will need to be resolved first. Would you be able to help fix those?

I agree this would be nice 😍 We need a bit of work to fix those issues before can add the pre-commit hook indeed.

**Comment by rijkvandermeulen** [comment]
@hrzn @gdevos010  makes sense! I can have a look at the existing mypy errors somewhere in the coming weeks.

**Comment by madtoinou** [comment]
Fixed by #1186.
